platform/ui/src/utils/ModalProvider.js

```diff
@@ -0,0 +1,91 @@
+import React, { useState, createContext, useContext } from 'react';
+import Modal from 'react-bootstrap-modal';
+
+const ModalContext = createContext(null);
+const { Provider, Consumer } = ModalContext;
+
+export const useModalContext = () => useContext(ModalContext);
+
+const ModalProvider = ({ children }) => {
+  const DEFAULT_OPTIONS = {
+    component: null,
+    backdrop: false,
+    keyboard: false,
+    show: true,
+    large: true,
+    closeButton: true,
+    title: null,
+    customClassName: null,
+  };
+
+  const [options, setOptions] = useState(DEFAULT_OPTIONS);
+
+  const show = (component, props = {}) =>
+    setOptions({
+      ...options,
+      component,
+      ...props,
+    });
+
+  const hide = () => setOptions(DEFAULT_OPTIONS);
+
+  return (
+    <Provider value={{ ...options, show, hide }}>
```

Comment:
Still a little new to React. Is it possible to use `useContext` here for the ModalContext instead of a provider? If both are possible, which one is preferable? Why?

Reply:
So, let me try to explain... The provider passes the show function to all children components, the children then call the show method with a modal component as argument, but the Modal itself (inside the provider) also needs the Modal context which have a component state that the user just set (the new component), so the modal consumes the same context to show the content. This is why theres a ModalContext being used inside the provider itself. The provider is required in order to create the context, and the context can be accessed through props or through the useContext.